server/restapi/restapi.coffee

```diff
@@ -0,0 +1,59 @@
+Api = new Restivus
+	useDefaultAuth: true
+	prettyJson: true
+
+
+Api.addCollection ChatMessage
+Api.addCollection ChatRoom
+
+
+Api.addRoute 'version', authRequired: false,
+	get: ->
+		version = {api: '0.1', rocketchat: '0.5'}
+		status: 'success', versions: version
+
+Api.addRoute 'publicRooms', authRequired: true,
+	get: ->
+		rooms = ChatRoom.find({ t: 'c' }, { sort: { msgs:-1 } }).fetch()
+		if rooms
+			status: 'success', rooms: rooms
+		else
+			statusCode: 404
+			body: status: 'fail', message: 'Rooms not found'
+
+# join a room
+Api.addRoute 'rooms/:id/join', authRequired: true,
+	post: ->
+		Meteor.runAsUser this.userId, () =>
+			Meteor.call('joinRoom', @urlParams.id)
+		status: 'success'   # need to handle error
+
+
+# leave a room
+Api.addRoute 'rooms/:id/leave', authRequired: true,
+	post: ->
+		Meteor.runAsUser this.userId, () =>
+			Meteor.call('leaveRoom', @urlParams.id)
+		status: 'success'   # need to handle error
+
+
+# get messages in a room
+Api.addRoute 'rooms/:id/messages', authRequired: true,
+	get: ->
```

Comment:
'canAccessRoom' should be enough.
on the client you can't visit private groups you are not in or 1v1 rooms of other users.. so we have to do some validation..
maybe we should check for an admin user/token if we want to violate this rules, but 'canAcessRoom' does not do it atm.